### DISCUSSION TITLE
My rule does not work, any way to debug?

### DISCUSSION THREAD

**Discussion by JPT77** [arendst/Tasmota#15083] [Q&A] [answered]
Hi, 

I've got a door switch (well four for cabinet doors) that should switch on lights during the night while any door is open.

I added a rule that depends on daylight timers. 

	PowerOnState 0
	BackLog Fade 1; Speed 5 Dimmer 10
	SwitchMode 2
	GPIO3 Switch 1
	Rule1 on Clock#Timer=timer1 do SwitchMode1 15 endon on Clock#Timer=timer2 do SwitchMode1 2 endon
	Rule1 1 

    {"Timer1":{"Enable":1,"Mode":1,"Time":"00:00","Window":0,"Days":"1111111","Repeat":1,"Output":1,"Action":3}}
    {"Timer2":{"Enable":1,"Mode":0,"Time":"12:10","Window":0,"Days":"1111111","Repeat":1,"Output":1,"Action":3}}
    {"Rule1":{"State":"ON","Once":"OFF","StopOnError":"OFF","Length":90,"Free":421,"Rules":"on Clock#Timer=timer1 do SwitchMode1 15 endon on Clock#Timer=timer2 do SwitchMode1 2 endon"}}

I 've set timer 2 to 12:10 for testing but just nothing happens.
Alternating switchmode between 2 and 15 works just fine for disabling the switch temporarily. (I would prefer an OFF-only mode, but that does not exist?)

thanks for any help

ps: Hardware H801, stock tasmota 11.0

**Comment by sfromis** [comment, marked as answer]
This can't work: `on Clock#Timer=timer1` - [should be](https://tasmota.github.io/docs/Rules/#examples-of-available-triggers) `on Clock#Timer=1`

What you posted does not indicate if you enabled the timers feature as such, not just the individual timers you posted. In the Web UI, there is an extra check box at the top for that.

**Reply by JPT77** [reply]
thanks, that did it.
